lib/data/wrapper.js

```diff
@@ -696,6 +715,19 @@ const data = {
             });
         }
 
+        if (locationTypeMatchGCP && dataLocator.lenegth === 1) {
```

Comment:
typo: `lenegth`

Maybe one more test to add? :wink:

Otherwise I'm not strongly against having this block copied from the AWS equivalent above, though here apart from the use of `gcpSourceKey` it seems to do pretty much the same. Is there an easy factorization here?

Reply:
will make the change to factorize this